docs/source/write_first_app.rst

```diff
@@ -101,889 +97,485 @@ bringing up a new one.
+In addition to the above, a hash implementation can also be specified. The default implementation will typically be
+sufficient.
 
-.. code:: bash
+The sample application uses an X.509 certificate for the Org1 user identity, and a signing implementation based on the
+corresponding private key. These credentials are created using the Org1 Certificate Authority when the test network
+is started.
 
-  Successfully registered and enrolled user appUser and imported it into the wallet
+.. code-block:: TypeScript
 
-Third, the sample application prepares a connection to the channel and smart contract
--------------------------------------------------------------------------------------
+  const client = await newGrpcConnection();
 
-In the prior steps, the application generated the admin and app user credentials and placed them in the wallet.
-If the credentials exist and have the correct permissions attributes associated with them, the sample application user
-will be able to call chaincode functions after getting reference to the channel name and contract name.
+  const gateway = connect({
+      client,
+      identity: await newIdentity(),
+      signer: await newSigner(),
+  });
 
-.. note:: Our connection configuration specifies only the peer from your own Org.
-          We tell node client sdk to use the service discovery (running on the peer),
-          which fetches other peers that are currently online, metadata like relevant endorsement policies
-          and any static information it would have otherwise needed to communicate with the rest of the nodes.
-          The ``asLocalhost`` set to ``true`` tells it to connect as localhost, since our client is running on same network as the other fabric nodes.
-          In deployments where you are not running the client on the same network as the other fabric nodes,
-          the ``asLocalhost`` option would be set to ``false``.
+  async function newIdentity(): Promise<Identity> {
+      const credentials = await fs.readFile(certPath);
+      return { mspId: 'Org1MSP', credentials };
+  }
 
-You will notice that in the following lines of application code, the application is getting reference
-to the Contract using the contract name and channel name via Gateway:
+  async function newSigner(): Promise<Signer> {
+      const privateKeyPem = await fs.readFile(keyPath);
+      const privateKey = crypto.createPrivateKey(privateKeyPem);
+      return signers.newPrivateKeySigner(privateKey);
+  }
 
-.. code:: bash
+Third, the application accesses the smart contract to be invoked
+----------------------------------------------------------------
 
-  // Create a new gateway instance for interacting with the fabric network.
-  // In a real application this would be done as the backend server session is setup for
-  // a user that has been verified.
-  const gateway = new Gateway();
+The sample application uses the Gateway connection to get a reference to the Network and then the default Contract
```

Comment:
so docs equate "chaincode" with "smart contract" elsewhere, but favor "chaincode". I guess you're saying the app references the smart contract within the chaincode, not the chaincode per se.

Reply:
There was a big discussion about this several years ago and a push to move the terminology away from the Fabric implementation-centric language to something that make more sense for end users. So the terminology of the Fabric programming model, exposed in the client APIs since v1.4, is Network (corresponding to channel) and Smart Contract. A chaincode package is what you deploy, and that package may contain multiple smart contracts. Often chaincodes only contain a single, default smart contract, in which case there is a one-to-one relationship between chaincode and smart contract. Partly because of this and partly for historic reasons, chaincode and smart contract often get used interchangeably. The world state is namespaced at the chaincode level, so all smart contracts within a given chaincode share the same world state.